File: Python3/can_place_flowers.py
class Solution:
    def canPlaceFlowers(self, flowerbed: List[int], n: int) -> bool:
        count = 0
        flowers = 0
        
        if flowerbed[0] == 0: flowerbed.insert(0, 0)
        if flowerbed[-1] == 0: flowerbed.append(0)

        for i in range(len(flowerbed)+1):
            if i < len(flowerbed) and flowerbed[i] == 0: 
                count += 1
            else:
                k = 0 if (count-1)//2 < 0 else (count-1)//2
                flowers += k
                count = 0
                
        return flowers >= n

File: C/can_place_flowers.c
bool canPlaceFlowers(int* flowerbed, int flowerbedSize, int n){
    int count = 0, flowers = 0, final = 0;
        
    if(*(flowerbed+0) == 0) count++;
    if(*(flowerbed+(flowerbedSize-1)) == 0) final++;

    for(int i = 0 ; i <= flowerbedSize; i++) {
        if(i < flowerbedSize && *(flowerbed+i) == 0)
            count ++;
        else {
            if(i == flowerbedSize) count += final;  
            flowers += (count-1)/2;
            count = 0;
        }
    }
                
    return flowers >= n;
}
